### PR TITLE
initial structure

### PR DESCRIPTION
**Pull request by MariaSarnova**

### CODE REVIEW

File: .idea/compiler.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (FilippStankevichTR) [inline .idea/compiler.xml:1]: do not commit anything under .idea or target. and do not merge your PR before review

File: src/main/java/com/epam/task/first/logic/ArrayLogic.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        for(Integer element : elements){
+            if(min>element){
+                min=element;
+            }
+        }
+
+        return min;
+    }
+
+    public List<Integer> replaceNonzeroElementsByOnesWithSign(Array array){
```
Comment (FilippStankevichTR) [inline src/main/java/com/epam/task/first/logic/ArrayLogic.java:43]: return new Array object

File: src/main/java/com/epam/task/first/logic/ArrayLogic.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        List<Integer> oldElements =array.getElements();
+
+        if(oldElements.size()==0){
+            throw new IllegalArgumentException("Array is empty");
+        }
+
+        List<Integer> newElements = new ArrayList<Integer>();
+        for(Integer element :oldElements){
+            if(element<0){
+                newElements.add(-1);
```
Comment (FilippStankevichTR) [inline src/main/java/com/epam/task/first/logic/ArrayLogic.java:53]: probably break here for performance

File: src/main/java/com/epam/task/first/logic/ArrayLogic.java [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+            }
+        }
+
+        return newElements;
+    }
+
+    public double findTheAverageOfTheElements(Array array){
+        List<Integer> elements =array.getElements();
+
+        if(elements.size()==0){
```
Comment (FilippStankevichTR) [inline src/main/java/com/epam/task/first/logic/ArrayLogic.java:71]: not sure if we need this check now, also don't use runtime exception now

File: src/main/java/com/epam/task/first/logic/ArrayLogic.java [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+            throw new IllegalArgumentException("Array is empty");
+        }
+
+        double average = 0;
```
Comment (FilippStankevichTR) [inline src/main/java/com/epam/task/first/logic/ArrayLogic.java:75]: define average where you calculate it

File: src/test/java/com/epam/task/first/logic/ArrayLogicTest.java [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+
+        Assert.assertEquals(expected,actual);
+    }
+
+    @Test
+    public void replaceNonzeroElementsByOnesWithSignShouldReplaceNonzeroElementsByMinusOnesWhenNegativeNumbersApplied(){
+        Array array = new Array(-1,-2,-4,-6,-20);
+        List<Integer> expected = new Array(-1,-1,-1,-1,-1).getElements();
+
+        List<Integer> actual = arrayLogic.replaceNonzeroElementsByOnesWithSign(array);
```
Comment (FilippStankevichTR) [inline src/test/java/com/epam/task/first/logic/ArrayLogicTest.java:83]: if you define method equals in Array class, and from your method you would return Array object, so you can use assertEquals

File: .idea/compiler.xml (hunk 1 above)
Comment (MariaSarnova) [reply .idea/compiler.xml:1]: create a new PR link:
https://github.com/MariaSarnova/FirstTask/pull/2

**Pull request merged**